base_layer/wallet/src/transaction_service/protocols/transaction_receive_protocol.rs

```diff
@@ -406,10 +250,75 @@ where TBackend: TransactionBackend + Clone + 'static
             },
         };
 
+        // Determine the time remaining before this transaction times out
+        let elapsed_time = Utc::now()
+            .naive_utc()
+            .signed_duration_since(inbound_tx.timestamp)
+            .to_std()
+            .map_err(|_| {
+                TransactionServiceProtocolError::new(
+                    self.id,
+                    TransactionServiceError::ConversionError("duration::OutOfRangeError".to_string()),
+                )
+            })?;
+
+        let timeout_duration = match self
+            .resources
+            .config
+            .pending_transaction_cancellation_timeout
+            .checked_sub(elapsed_time)
+        {
+            None => {
+                // This will cancel the transaction and exit this protocol
+                return self.timeout_transaction().await;
+            },
+            Some(t) => t,
+        };
+        let mut timeout_delay = delay_for(timeout_duration).fuse();
+
+        // check to see if a resend is due
+        let resend = match inbound_tx.last_send_timestamp {
+            None => true,
+            Some(timestamp) => {
+                let elapsed_time = Utc::now()
+                    .naive_utc()
+                    .signed_duration_since(timestamp)
+                    .to_std()
+                    .map_err(|_| {
+                        TransactionServiceProtocolError::new(
+                            self.id,
+                            TransactionServiceError::ConversionError("duration::OutOfRangeError".to_string()),
+                        )
+                    })?;
+                elapsed_time > self.resources.config.transaction_resend_period
+            },
+        };
+
+        if resend {
+            if let Err(e) = send_transaction_reply(
+                inbound_tx.clone(),
+                self.resources.outbound_message_service.clone(),
+                self.resources.config.direct_send_timeout,
+            )
+            .await
+            {
+                warn!(
+                    target: LOG_TARGET,
+                    "Error resending Transaction Reply (TxId: {}): {:?}", self.id, e
+                );
+            }
+            self.resources
+                .db
+                .increment_send_count(self.id)
+                .await
+                .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
```

Comment:
I considered it as an attempt was made to send and we will only try again (succeed or fail) again in the next period. Wanted to be careful not to get stuck attempting too many sends as it would thrash a mobile's battery.